extension/history.js

```diff
@@ -20,107 +20,110 @@ export class Database {
         const database = e.target.result;
         resolve(database);
       };
-      request.onerror = function(e) {
-        reject("Error: failed to create a database!");
+      request.onerror = e => {
+        reject(new Error(`Failed to create a database: ${e.target.errorCode}`));
       };
     });
   }
 
   get(primaryKey, TBName) {
     return new Promise((resolve, reject) => {
-      const request = indexedDB.open(this.DBName);
+      const request = indexedDB.open(this.dbName);
       request.onsuccess = e => {
         const database = e.target.result;
         const transaction = database.transaction([TBName]);
         const objectStore = transaction.objectStore(TBName);
         const read = objectStore.get(primaryKey);
-
-        read.onerror = function(event) {
-          reject("Unable to retrieve data from database!");
-        };
-
-        read.onsuccess = function(event) {
-          log.info("get result:", read.result);
+        read.onsuccess = e => {
+          log.info("Got result:", read.result);
           resolve(read.result);
         };
+        read.onerror = e => {
+          reject(new Error(`Unable to retrieve data from database: ${e.target.errorCode}`));
+        };
         database.close();
       };
-      request.onerror = event => reject("Unable to retrieve from database!");
+      request.onerror = e => {
+        reject(new Error(`Unable to retrieve from database: ${e.target.errorCode}`));
+      };
     });
   }
 
   getAll(TBName) {
-    return new Promise(resolve => {
-      const request = indexedDB.open(this.DBName);
+    return new Promise((resolve, reject) => {
+      const request = indexedDB.open(this.dbName);
       request.onsuccess = e => {
         const database = e.target.result;
         const objectStore = database.transaction(TBName).objectStore(TBName);
-        const list = [];
```

Comment:
@OliviaRuyinZhang Changed this to use getAll(), according to this

![image](https://user-images.githubusercontent.com/24581402/76932072-71c58100-68c0-11ea-9626-4983d0bc3a78.png)

Let me know if you have comments on this, i.e., if you prefer the way you did it.